examples/pytorch/seal_ogbl/main.py

```diff
@@ -0,0 +1,426 @@
+        test_hits = evaluator.eval({
+            'y_pred_pos': pos_test_pred,
+            'y_pred_neg': neg_test_pred,
+        })[f'hits@{K}']
+
+        results[f'Hits@{K}'] = (valid_hits, test_hits)
+
+    return results
+     
+
+def evaluate_mrr(pos_val_pred, neg_val_pred, pos_test_pred, neg_test_pred):
+    print(pos_val_pred.size(), neg_val_pred.size(), pos_test_pred.size(), neg_test_pred.size())
+    neg_val_pred = neg_val_pred.view(pos_val_pred.shape[0], -1)
+    neg_test_pred = neg_test_pred.view(pos_test_pred.shape[0], -1)
+    results = {}
+    valid_mrr = evaluator.eval({
+        'y_pred_pos': pos_val_pred,
+        'y_pred_neg': neg_val_pred,
+    })['mrr_list'].mean().item()
+
+    test_mrr = evaluator.eval({
+        'y_pred_pos': pos_test_pred,
+        'y_pred_neg': neg_test_pred,
+    })['mrr_list'].mean().item()
+
+    results['MRR'] = (valid_mrr, test_mrr)
+    
+    return results
+
+
+def evaluate_auc(val_pred, val_true, test_pred, test_true):
+    valid_auc = roc_auc_score(val_true, val_pred)
+    test_auc = roc_auc_score(test_true, test_pred)
+    results = {}
+    results['AUC'] = (valid_auc, test_auc)
+
+    return results
+
+
+if __name__ == '__main__':
+    # Data settings
+    parser = argparse.ArgumentParser(description='OGBL (SEAL)')
+    parser.add_argument('--dataset', type=str, default='ogbl-collab')
+    # GNN settings
+    parser.add_argument('--model', type=str, default='DGCNN')
+    parser.add_argument('--sortpool_k', type=float, default=0.6)
+    parser.add_argument('--num_layers', type=int, default=3)
+    parser.add_argument('--hidden_channels', type=int, default=32)
+    parser.add_argument('--batch_size', type=int, default=32)
+    # Subgraph extraction settings
+    parser.add_argument('--num_hops', type=int, default=1)
+    parser.add_argument('--ratio_per_hop', type=float, default=1.0)
+    parser.add_argument('--use_feature', action='store_true', 
+                        help="whether to use raw node features as GNN input")
+    parser.add_argument('--use_edge_weight', action='store_true', 
+                        help="whether to consider edge weight in GNN")
+    # Training settings
+    parser.add_argument('--lr', type=float, default=0.0001)
+    parser.add_argument('--epochs', type=int, default=50)
+    parser.add_argument('--runs', type=int, default=1)
+    parser.add_argument('--train_percent', type=float, default=100)
+    parser.add_argument('--val_percent', type=float, default=100)
+    parser.add_argument('--test_percent', type=float, default=100)
+    parser.add_argument('--dynamic_train', action='store_true', 
+                        help="dynamically extract enclosing subgraphs on the fly")
+    parser.add_argument('--dynamic_val', action='store_true')
+    parser.add_argument('--dynamic_test', action='store_true')
+    parser.add_argument('--num_workers', type=int, default=16, 
+                        help="number of workers for dynamic mode; will be set to 0 if not dynamic")
```

Comment:
Dataloaders hang on machines without sufficient CPU capacity. Perhaps you should change the default value to 0 and leave a comment in the README file.